src/tarski/syntax/arithmetic/__init__.py

```diff
@@ -38,3 +69,148 @@ def summation( *args ):
+    return sort.language.constant(1, sort)
+
+
+def simplify(expr: Term):
+    if isinstance(expr, Constant):
+        return expr
+    elif isinstance(expr, Variable):
+        return expr
+    elif isinstance(expr, CompoundTerm):
+        if not expr.symbol.builtin :
+            return expr
+        if expr.symbol.symbol == BuiltinFunctionSymbol.ADD:
+            simp_st = (simplify(expr.subterms[0]), simplify(expr.subterms[1]))
+            if simp_st[0].is_syntactically_equal(zero(expr.sort)):
+                return simp_st[1]
+            if simp_st[1].is_syntactically_equal(zero(expr.sort)):
+                return simp_st[0]
+            expr.subterms = simp_st
+            return expr
+        if expr.symbol.symbol == BuiltinFunctionSymbol.SUB:
+            simp_st = (simplify(expr.subterms[0]), simplify(expr.subterms[1]))
+            if simp_st[0].is_syntactically_equal(zero(expr.sort)):
+                return -1 * simp_st[1]
+            if simp_st[1].is_syntactically_equal(zero(expr.sort)):
+                return simp_st[0]
+            expr.subterms = simp_st
+            return expr
+        if expr.symbol.symbol == BuiltinFunctionSymbol.MUL:
+            simp_st = (simplify(expr.subterms[0]), simplify(expr.subterms[1]))
+            if simp_st[0].is_syntactically_equal(zero(expr.sort)):
+                return zero(expr.sort)
+            if simp_st[1].is_syntactically_equal(zero(expr.sort)):
+                return zero(expr.sort)
+            if simp_st[0].is_syntactically_equal(one(expr.sort)):
+                return simp_st[1]
+            if simp_st[1].is_syntactically_equal(one(expr.sort)):
+                return simp_st[0]
+            expr.subterms = simp_st
+            return expr
+        if expr.symbol.symbol == BuiltinFunctionSymbol.DIV:
+            simp_st = (simplify(expr.subterms[0]), simplify(expr.subterms[1]))
+            if simp_st[0].is_syntactically_equal(zero(expr.sort)):
+                return zero(expr.sort)
+            if simp_st[1].is_syntactically_equal(zero(expr.sort)):
+                raise err.SemanticError("Division by Zero detected when \
+            simplifying expression '{}' in division '{}'".format(expr.subterms[1], expr))
+            if simp_st[1].is_syntactically_equal(one(expr.sort)):
+                return simp_st[0]
+            expr.subterms = simp_st
+            return expr
+        if expr.symbol.symbol == BuiltinFunctionSymbol.EXP:
+            simp_st = simplify(expr.subterms[0])
+            if simp_st.is_syntactically_equal(zero(expr.sort)):
+                return one(expr.sort)
+            expr.subterms = (simp_st,)
+            return expr
+    elif isinstance(expr, Matrix) or isinstance(expr, np.matrix):
+        N, M = expr.shape
+        for i in range(N):
+            for j in range(M):
+                expr[i,j] = simplify(expr[i,j])
+        return expr
+    elif isinstance(expr, AggregateCompoundTerm):
+        expr.subterm = simplify(subterm)
```

Comment:
subterm is not defined here

Reply:
Fixed in commit 73c80504e9800f30b36ad4c9ae5fdc3d2900d271